livingatlas/pipelines/src/main/java/au/org/ala/pipelines/beam/ALAUUIDValidationPipeline.java

```diff
@@ -0,0 +1,335 @@
+    String[] combinedArgs = new CombinedYamlConfiguration(args).toArgs("general", "uuid");
+    UUIDPipelineOptions options =
+        PipelinesOptionsFactory.create(UUIDPipelineOptions.class, combinedArgs);
+    MDC.put("datasetId", options.getDatasetId());
+    MDC.put("attempt", options.getAttempt().toString());
+    MDC.put("step", "VALIDATE_UUID");
+    PipelinesOptionsFactory.registerHdfs(options);
+    run(options);
+    // FIXME: Issue logged here: https://github.com/AtlasOfLivingAustralia/la-pipelines/issues/105
+    System.exit(0);
+  }
+
+  public static void run(UUIDPipelineOptions options) throws Exception {
+
+    Pipeline p = Pipeline.create(options);
+
+    // deletePreviousValidation
+    deletePreviousValidation(options);
+
+    // validation results
+    PCollectionList<String> results = PCollectionList.<String>empty(p);
+
+    ALAPipelinesConfig config =
+        ALAPipelinesConfigFactory.getInstance(
+                options.getHdfsSiteConfig(), options.getCoreSiteConfig(), options.getProperties())
+            .get();
+
+    // create key value store for data resource metadata
+    KeyValueStore<String, ALACollectoryMetadata> dataResourceKvStore =
+        ALAAttributionKVStoreFactory.create(config);
+
+    // lookup collectory metadata for this data resource
+    ALACollectoryMetadata collectoryMetadata = dataResourceKvStore.get(options.getDatasetId());
+    if (collectoryMetadata.equals(ALACollectoryMetadata.EMPTY)) {
+      log.error("Unable to retrieve dataset metadata for dataset: " + options.getDatasetId());
+      PCollection<String> pc =
+          p.apply(Create.of(METADATA_AVAILABLE + ": false").withCoder(StringUtf8Coder.of()));
+      results = results.and(pc);
+    } else {
+      PCollection<String> pc =
+          p.apply(Create.of(METADATA_AVAILABLE + ": true").withCoder(StringUtf8Coder.of()))
+              .setCoder(StringUtf8Coder.of());
+      ;
+      results = results.and(pc);
+    }
+
+    List<String> uniqueTerms = Collections.emptyList();
+
+    // construct unique list of darwin core terms
+    if (collectoryMetadata.getConnectionParameters() != null) {
+      uniqueTerms = collectoryMetadata.getConnectionParameters().getTermsForUniqueKey();
+      if ((uniqueTerms == null || uniqueTerms.isEmpty())) {
+        log.error(
+            "Unable to proceed, No unique terms specified for dataset: " + options.getDatasetId());
+        PCollection<String> pc =
+            p.apply(Create.of(UNIQUE_TERMS_SPECIFIED + ": false").withCoder(StringUtf8Coder.of()));
+        results = results.and(pc);
+      } else {
+        PCollection<String> pc =
+            p.apply(Create.of(UNIQUE_TERMS_SPECIFIED + ": true").withCoder(StringUtf8Coder.of()));
+        results = results.and(pc);
+      }
+    } else {
+      PCollection<String> pc =
+          p.apply(Create.of(UNIQUE_TERMS_SPECIFIED + ": false").withCoder(StringUtf8Coder.of()));
+      results = results.and(pc);
+    }
```

Comment:
pc/result code repeated.

Reply:
this has been refactored to remove the duplication